src/ng-pattern-restrict.js

```diff
@@ -238,10 +230,10 @@ angular.module('ngPatternRestrict', [])
 
           function detectSetCaretPositionMethods() {
             var input = iElement[0];
-            if (notThrows(function () { input.setSelectionRange(0, 0); })) {
+            if (typeof(input.setSelectionRange) === 'function') {
               setCaretPosition = setCaretPositionWithSetSelectionRange;
-            } else if (notThrows(function () { return input.createTextRange(); })) {
```

Comment:
Ok, after checking I was not able to find out which was the browser that was doing this. It may be that I did something wrong -- which would be great because I would like to simplify that code like you did.

I have not tested in old versions of Opera as I did before.

I'll continue reviewing the rest of the changes.